pandas/core/frame.py

```diff
@@ -5070,14 +5070,38 @@ def notna(self) -> DataFrame:
     def notnull(self) -> DataFrame:
         return ~self.isna()
 
+    # Overload function signature to prevent union of conflicting types
+    # As Optional[DataFrame] is really Union[DataFrame, None]
+    @overload
+    def dropna(
+        self,
+        axis: Axis = ...,
+        how: str = ...,
+        thresh: Optional[int] = ...,
+        subset: Optional[Union[Hashable, Sequence[Hashable]]] = ...,
+        inplace: Literal[False] = False,
```

Comment:
should you be specifying the default value here? (I'm not sure)

Reply:
Looks like we were typing at the time :P My understanding of overloading (and this might be very wrong) is that you only specify the defaults in overloads for those parameters you are overloading, in this case `inplace`.

See below for an example taken from `pandas.core.frame.reset_index`. 
https://github.com/pandas-dev/pandas/blob/67d4cae17bec45e84b9cf51bcf4fb5bbe293b26f/pandas/core/frame.py#L4812-L4843

Reply:
yeah in this example the defaults are not specified in the overloaded signatures so I think you should get rid of those

Reply:
That example also explains the mypy error (it's caused to a mypy bug). So to get mypy passing you need to add a type ignore (and a comment with the link to mypy issue and the error message above the overloaded signature as done in the `reset_index` example)

Those things being done I think this will be good to go in

Reply:
Gotcha! Changes forthcoming...